create_graphics.py

```diff
@@ -56,11 +64,48 @@ def create_maps(cla, fhr, grib_path, workdir):
                     msg = f'graphics: {variable} {level}'
                     raise errors.NoGraphicsDefinitionForVariable(msg)
 
-                args.append((fhr, grib_path, level, spec,
+                args.append((fhr, gribfile.contents, level, spec,
                              variable, workdir, tile))
 
-    with Pool(processes=cla.nprocs) as pool:
-        pool.starmap(parallel_maps, args)
+        print(f'Queueing {len(args)} maps')
+        with Pool(processes=cla.nprocs) as pool:
+            pool.starmap(parallel_maps, args)
+
+
+def create_zip(png_files, zipf):
+
+    ''' Create a zip file. Use a locking mechanism -- write a lock file to disk. '''
+
+    lock_file = f'{zipf}._lock'
+    retry = 2
+    count = 0
+    while True:
+        if not os.path.exists(lock_file):
+            fd = open(lock_file, 'w')
+            print(f'Writing to zip file for files like: {png_files[0][-10:]}')
+
+            try:
+                with zipfile.ZipFile(zipf, 'a', zipfile.ZIP_DEFLATED) as zfile:
+                    for png_file in png_files:
+                        if os.path.exists(png_file):
+                            zfile.write(png_file, os.path.basename(png_file))
+            except: # pylint: disable=bare-except
+                print(f'Error on writing zip file! {sys.exc_info()[0]}')
+                count += 1
+                if count >= retry:
+                    raise
+            else:
+                # When zipping is successful, remove png_files
+                for png_file in png_files:
+                    if os.path.exists(png_file):
+                        os.remove(png_file)
```

Comment:
This block executes if `try` block was successful. Only remove the png files then.